src/eth_provider/pending_pool.rs

```diff
@@ -1,17 +1,43 @@
 use crate::eth_provider::provider::EthDataProvider;
+use lazy_static::lazy_static;
+use std::str::FromStr;
+use std::time::Instant;
 use tokio::time::{sleep, Duration};
 
+lazy_static! {
+    // Interval between retries of transactions (in seconds)
+    pub static ref RETRY_TX_INTERVAL: usize = usize::from_str(
+        &std::env::var("RETRY_TX_INTERVAL")
+            .unwrap_or_else(|_| panic!("Missing environment variable RETRY_TX_INTERVAL"))
+    ).expect("failing to parse RETRY_TX_INTERVAL");
+}
+
 pub async fn start_retry_service<SP>(eth_provider: EthDataProvider<SP>)
 where
     SP: starknet::providers::Provider + Send + Sync,
 {
+    // Initialize last print time
+    let mut last_print_time = Instant::now();
+
     // Start an infinite loop.
     loop {
+        // Measure start time
+        let start_time_fn = Instant::now();
         // Call the retry_transactions method
         if let Err(err) = eth_provider.retry_transactions().await {
             tracing::error!("Error while retrying transactions: {:?}", err);
         }
-        // 30-second pause
-        sleep(Duration::from_secs(30)).await;
+        // Calculate elapsed time in milliseconds
+        let elapsed_time_ms = start_time_fn.elapsed().as_millis();
+
+        // Check if 5 minutes have passed since the last print
+        if last_print_time.elapsed() >= Duration::from_secs(300) {
+            println!("Elapsed time to retry transactions (milliseconds): {}", elapsed_time_ms);
```

Comment:
We should use the tracing crate instead of prints

Reply:
cc @greged93 maybe we can use prometheus instead? or since this is internal we cant

Reply:
- Just fixed about tracing
- As this is just a temp display that we will remove once we have more information about times, don't know if this is needed no?

Reply:
I don't think prometheus can help, but let me check